### DISCUSSION TITLE
Can Scriptkit do this?

### DISCUSSION THREAD

**Discussion by stevebarakat** [johnlindquist/kit#423] [Q&A] [answered]
I use Create React App a lot. A few times a day sometimes just to try things out. Every time I start a new CRA I do the exact same thing every time. I delete the same files, update the same files, and add the same boilerplate... every time.

Could I use Scriptkit to automate that process... possibly? That would be fantastic.

**Comment by johnlindquist** [comment, marked as answer]
Yeah, for sure!

Can you post the sequence of commands you use? I can toss the rest together pretty easily to get you started

**Comment by stevebarakat** [comment]
Oh. I don't use commands. I do it by hand. That's my point.

I'm very front-end. 

I know how to make a basic CRUD app in Express. That's about as far as I go with Node.js. I don't have any experience using the file system really. Do I need to know Node.js to do what I want to do? I could probably figure it out. Scriptkit might be a good way to start learning more about that side of things. I actually started reading the error log I sent you and was quite intrigued.

**Reply by johnlindquist** [reply]
Does "by hand" mean in Finder? I'm just looking for a step-by-step of your process. Which folder you use, which files you update/remove, etc. Just as specific as possible so I can point you on the right path.

**Reply by stevebarakat** [reply]
There's two different development environments I'd like to automate. One for permanent projects (with Storybook, Testing Library, and Typescript) and one for sandboxes (with just React and styled-components). But the sandbox will be used more often so that's the one I'll start with.

I made a screen-clip of the setup:

https://user-images.githubusercontent.com/3830326/132069061-556a650f-3701-491d-8c85-e14bbb3f480d.mov

— I delete the default content, assets, styles, and tests. 
— I rename index.css to reset.css and replace the contents with the reset below (Eric Meyer Reset + a box-sizing rule). I keep the App.css file also but delete all it's contents.
— I uninstall web-vitals and install styled-components. 
— That's it!

/* http://meyerweb.com/eric/tools/css/reset/ 
    v2.0 | 20110126
    License: none (public domain)
*/

html, body, div, span, applet, object, iframe,
h1, h2, h3, h4, h5, h6, p, blockquote, pre,
a, abbr, acronym, address, big, cite, code,
del, dfn, em, img, ins, kbd, q, s, samp,
small, strike, strong, sub, sup, tt, var,
b, u, i, center,
dl, dt, dd, ol, ul, li,
fieldset, form, label, legend,
table, caption, tbody, tfoot, thead, tr, th, td,
article, aside, canvas, details, embed, 
figure, figcaption, footer, header, hgroup, 
menu, nav, output, ruby, section, summary,
time, mark, audio, video {
	margin: 0;
	padding: 0;
	border: 0;
	font-size: 100%;
	font: inherit;
	vertical-align: baseline;
}
/* HTML5 display-role reset for older browsers */
article, aside, details, figcaption, figure, 
footer, header, hgroup, menu, nav, section {
	display: block;
}
body {
	line-height: 1;
}
ol, ul {
	list-style: none;
}
blockquote, q {
	quotes: none;
}
blockquote:before, blockquote:after,
q:before, q:after {
	content: '';
	content: none;
}
table {
	border-collapse: collapse;
	border-spacing: 0;
}
*,
*::before,
*::after {
  box-sizing: border-box;
}

**Comment by JoepKockelkorn** [comment]
While scriptkit can definitely be used to achieve this, I think [(angular) schematics](https://angular.io/guide/schematics) is built specifically for this 'scaffolding code' job you describe. You can dry-run a schematic to see if it has the desired effect and it's also atomic, meaning when a substep fails, the whole fails. It does have a somewhat steep learning curve, but reading and parsing files (using ast tooling?) has to be done anyway regardless of tooling choice.

**Reply by JoepKockelkorn** [reply]
Ok maybe parsing and updating within a file is not what you described but it could be a requirement that is right around the corner.

**Reply by stevebarakat** [reply]
Okay. I think I can figure it out. Thank for the help!
